src/server/physics/world.js

```diff
@@ -1,149 +1,230 @@
 module.exports = function(Ammo, config) {
-	// Physics letiables
-	let collisionConfiguration,
-		dispatcher,
-		broadphase,
-		solver,
+	let _collisionConfiguration,
+		_dispatcher,
+		_broadphase,
+		_solver,
 		physicsWorld;
 
 	// Physics configuration
-	collisionConfiguration = new Ammo.btDefaultCollisionConfiguration();
-	dispatcher = new Ammo.btCollisionDispatcher( collisionConfiguration );
-	broadphase = new Ammo.btDbvtBroadphase();
-	solver = new Ammo.btSequentialImpulseConstraintSolver();
-	physicsWorld = new Ammo.btDiscreteDynamicsWorld( dispatcher, broadphase, solver, collisionConfiguration );
+	_collisionConfiguration = new Ammo.btDefaultCollisionConfiguration();
+	_dispatcher = new Ammo.btCollisionDispatcher(_collisionConfiguration );
+	_broadphase = new Ammo.btDbvtBroadphase();
+	_solver = new Ammo.btSequentialImpulseConstraintSolver();
+	physicsWorld = new Ammo.btDiscreteDynamicsWorld(_dispatcher, _broadphase, _solver, _collisionConfiguration );
 	physicsWorld.setGravity( new Ammo.btVector3( 0, config.physics.gravity, 0 ) );
 
-	function createTerrainShape() {
-		// Up axis = 0 for X, 1 for Y, 2 for Z. Normally 1 = Y is used.
-		let upAxis = 1;
+	let _lastPhysicsUpdate = Date.now();
 
-		// hdt, height data type. "PHY_FLOAT" is used. Possible values are "PHY_FLOAT", "PHY_UCHAR", "PHY_SHORT"
-		let hdt = "PHY_FLOAT";
+	function _updatePhysics( deltaTime ) {
+		physicsWorld.stepSimulation( deltaTime, 10 );
+	}
+
+	let _updateInterval;
+	let _startUpdateInterval = function() {
+		_updateInterval = setInterval(function() {
+			let now = Date.now();
+			let deltaTime = (now - _lastPhysicsUpdate) / 1000;
+			_lastPhysicsUpdate = now;
+			_updatePhysics(deltaTime);
+		}, 1000 / config.physics.steps);
+	};
 
-		// Set this to your needs (inverts the triangles)
-		let flipQuadEdges = false;
+	_startUpdateInterval();
+
+	return {
+		map: undefined,
+		physicsWorld,
+		updateInterval: undefined,
+		gates: [],
+
+		toggleGates(override) {
```

Comment:
I think this function could be a bit better named/used. Maybe `setGateStates` where the parameter could be `"open"` or `"close"` and is explicitly handled in both of these cases.